enkei: init at 0.9.3

ppenguin opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Artturin commented on pkgs/tools/wayland/enkei/default.nix:
```diff
@@ -0,0 +1,55 @@
+  buildInputs = [
+  ];
+
+  PKG_CONFIG_PATH = lib.strings.makeSearchPath "lib/pkgconfig" libdeps; # why is this necessary to do manually?! (thought it was automagic?)
```
buildRustPackage has strictDeps enabled to lib deps in nativeBuildInputs won't work.
```suggestion
  buildInputs = [ wayland libGL glibc cairo ];
```
this will work

ppenguin replied:
@Artturin sorry, completely forgot about this PR...

Artturin commented on pkgs/tools/wayland/enkei/default.nix:
```diff
@@ -0,0 +1,55 @@
+  nativeBuildInputs = [
+    pkg-config
+  ] ++ libdeps;
+  buildInputs = [
+  ];
```
```suggestion
```

Artturin commented on pkgs/tools/wayland/enkei/default.nix:
```diff
@@ -0,0 +1,53 @@
+    pkg-config
+  ] ++ libdeps;
+
+  buildInputs = [ wayland libGL glibc cairo ];
```
```suggestion
  buildInputs = [ wayland libGL glib cairo ];
```

Artturin commented on pkgs/tools/wayland/enkei/default.nix:
```diff
@@ -0,0 +1,53 @@
+
+rustPlatform.buildRustPackage rec {
+  pname = "enkei";
+  version = "v0.9.3";
```
```suggestion
  version = "0.9.3";
```

ppenguin committed:
enkei: init @ v0.9.3

Update pkgs/by-name/en/enkei/package.nix

Co-authored-by: Arne Keller <2012gdwu+github@posteo.de>

Update pkgs/by-name/en/enkei/package.nix

Co-authored-by: Arne Keller <2012gdwu+github@posteo.de>

enkei: init @ v0.9.3

Cleanup / remove top-level attr (since using pkgs/by-name)

nixfmt

Artturin reviewed (approved):
Removed `libs` from `nativeBuildInputs`, switched to `cargoHash` because vendoring lockfiles is bad for the git repo size, formatted with nixfmt, moved to `by-name`

FliegendeWurst commented on pkgs/by-name/en/enkei/package.nix:
```diff
@@ -0,0 +1,48 @@
+
+  src = fetchFromGitHub {
+    owner = "jwuensche";
+    repo = pname;
```
```suggestion
    repo = "enkei";
```
#277994

FliegendeWurst commented on pkgs/by-name/en/enkei/package.nix:
```diff
@@ -0,0 +1,48 @@
+    glib
+    cairo
+  ];
+
```
```suggestion

  doCheck = false; # no tests

```
